eng/testing/outerBuild.targets

```diff
@@ -0,0 +1,19 @@
+<Project>
+  <Target Name="Test">
+    <ChooseBestTargetFrameworksTask BuildTargetFrameworks="$(BuildTargetFramework)-$(BuildOS)"
+                                    SupportedTargetFrameworks="$(TargetFrameworks)"
+                                    RuntimeGraph="$(RuntimeGraph)" >
+      <Output TaskParameter="BestTargetFrameworks" ItemName="_BestTargetFramework" />
+    </ChooseBestTargetFrameworksTask>
+
+    <AddTargetFrameworksToProjectTask ProjectName="$(MSBuildProjectFile)"
+                                      BestTargetFrameworks="@(_BestTargetFramework)" >
+      <Output TaskParameter="InnerBuildProjects" ItemName="TestProject" />
+    </AddTargetFrameworksToProjectTask>
+    
+    <MSBuild Projects="@(TestProject)"
+             Targets="Test"
```

Comment:
@Anipik @ericstj how would this work with `dotnet test` where we don't invoke the custom `Test` target?

Reply:
@ViktorHofer can you give me a test case scenario where dotnet test works so that i can test the change of behavior ?

Reply:
We decided to take a look at how this works with dotnet test post merge.